### PR TITLE
Improve the preparation of the testing environment

### PR DESCRIPTION
**Pull request by ZhukovGreen**
- Instead of separate ansible playbooks the configurable ansible roles are used
- By adjusting ansible group_vars/all.yml it is possible to specify the rpm provider and build mechanism

The tmt-pytest framework documentation was improved and simplified

### CODE REVIEW

**Comment by ZhukovGreen** [comment]
@oamg/convert2rhel for a review

**Review by bocekm — CHANGES_REQUESTED** [review]
What's the purpose of the empty `defaults/main.yml` files?

File: tests/ansible_collections/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+- hosts: all
+  roles:
+
+    - role: oracle-linux-specific
+      when: ansible_facts['distribution'] == "OracleLinux"
+      tags:
+        - oracle
```
Comment (bocekm) [inline tests/ansible_collections/main.yml:8]: What's the purpose of the tags?

**Review by bocekm — COMMENTED** [review]
I've tried both the local and URL options and both work. Good job.

File: tests/ansible_collections/main.yml (hunk 1 above)
Comment (ZhukovGreen) [reply tests/ansible_collections/main.yml:8]: Similar to defaults in roles. Agree let's remove it unless we'll feel a need in the possibility to reuse roles for maunall runs. The tags allow running certain scenarios, i.e.
ansible-playbook --tags packaging will run only roles in sequence with tag packaging.

**Review by ZhukovGreen — COMMENTED** [review]
Fixed all comments @bocekm

File: tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+---
 - name: Install standard kernel
   yum:
     name: kernel
```
Comment (ZhukovGreen) [inline tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml:4]: Inna found an issue, with this task when her machine had kernel-transition package installed, this make this tasks skip to install the RHCK kernel. Should be changed to something like kernel.*

**Review by bocekm — APPROVED** [review]

File: tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml (hunk 2 above)
Comment (bocekm) [reply tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml:4]: We agreed on resolving this in a separate PR.

**Pull request merged**